src/gui/qgsmaptoolcapture.cpp

```diff
@@ -1391,6 +1391,29 @@ void QgsMapToolCapture::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
       }
       else
       {
+
+        //does compoundcurve contain circular strings?
+        //does provider support circular strings?
+        QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( layer() );
+        const bool isVectorLayer = ( vlayer != nullptr ) && ( vlayer->type() == Qgis::LayerType::Vector );
+        if ( isVectorLayer )
+        {
+          const bool hasCurvedSegments = captureCurve()->hasCurvedSegments();
+          const bool providerSupportsCurvedSegments = vlayer->dataProvider()->capabilities() & QgsVectorDataProvider::CircularGeometries;
+
+          if ( !isVectorLayer || ( hasCurvedSegments && providerSupportsCurvedSegments ) )
```

Comment:
Isn't `!isVectorLayer` always `false` here?

Reply:
`isVectorLayer` no longer exist now after bf8e32d0a83b562c2e263575af6f75daea852083

Reply:
yes, it's fixed